File: front/src/app/feature/admin-management/type/session/sessionDTO.ts
import {Category, Format, Reviews} from './session';

export interface SessionDTO {
  id?: string;
  title?: string;
  abstractText?: string;
  speakers?: Speaker[];
  createdAt?: string;
  deliberationStatus?: boolean;
  confirmationStatus?: boolean;
  level?: string;
  eventId?: string;
  formats?: Format[];
  categories?: Category[];
  tags?: string[];
  languages?: string[];
  references?: string;
  reviews?: Reviews;
}

export interface Speaker {
  name?: string;
  bio?: string;
  company?: string;
  references?: string;
  picture?: string;
  location?: string;
  email?: string;
  socialLinks?: string[];
}

File: back/src/main/java/com/speakerspace/dto/session/SessionDTO.java
package com.speakerspace.dto.session;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Builder;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

@Builder
public record SessionDTO (
        String id,
        String title,
        @JsonProperty("abstract")
        String abstractText,
        String deliberationStatus,
        String confirmationStatus,
        String level,
        String references,
        List<FormatDTO> formats,
        List<CategoryDTO> categories,
        List<String> tags,
        List<String> languages,
        List<SpeakerDTO> speakers,
        ReviewDTO reviews,
        String eventId,
        LocalDateTime start,
        LocalDateTime end,
        String track,
        Date createdAt,
        Date updatedAt
){}